react-stopwatch-component/src/stopwatch.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+export default class Stopwatch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isRunning: false, countDown: 0 };
+    this.timer = this.timer.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+
+    this.handleReset = this.handleReset.bind(this);
+  }
+
+  timer() {
+    this.intervalID = setInterval(() => {
+      this.setState({ countDown: this.state.countDown + 1 });
+    }, 1000);
+  }
+
+  handleClick() {
+    this.setState({ isRunning: !this.state.isRunning });
+    if (this.state.isRunning) {
+      clearInterval(this.intervalID);
+    } else {
+      this.timer();
+    }
```

Comment:
Setting state before the `if` statement is misleading, remember `setState` is asynchronous. It is better to do the `setState` at the end of the function.
```JAVASCRIPT
if (this.state.isRunning) {
  clearInterval(this.intervalID);
} else {
  this.timer();
}
this.setState({ isRunning: !this.state.isRunning });
```